[DataTable] Add optional totalsMarkupLabel prop and fixing Totals plurality (#2660)
Shopify/polaris-react: src/components/DataTable/README.md

}
```

### Data table with custom totals labels

> Use to pass in custom totals labels for both singular and plural amount of totals items. The plurality is based on the amount of items passed in for totals.

Reviewer: ```suggestion
Use to provide a custom heading for the totals row.
```

```jsx
function DataTableExample() {
  const rows = [['Emerald Silk Gown', '$875.00']];

  return (
    <Page title="Sales by product">
      <Card>
        <DataTable
          columnContentTypes={['text', 'text']}
          headings={['Products']}
          rows={rows}
          totals={['', '$155,830.00']}
          totalsMarkupLabel={{
            singular: 'Singular',

>             plural: 'Plural',

Reviewer: Is there a more realistic example that we can use here? Maybe from the use case that prompted these changes?

Author: Good idea, changing it to `Total` and `Totals`

          }}

>           showTotalsInFooter

Reviewer: ```suggestion
```
